app/serializers/listable_topic_serializer.rb

```diff
@@ -31,14 +31,18 @@ class ListableTopicSerializer < BasicTopicSerializer
   has_one :last_poster, serializer: BasicUserSerializer, embed: :objects
 
   def image_url
-    object.image_url
+    object.image_url(enqueue_if_missing: true)
   end
 
   def thumbnails
     extra_sizes = ThemeModifierHelper.new(request: scope.request).topic_thumbnail_sizes
     object.thumbnail_info(enqueue_if_missing: true, extra_sizes: extra_sizes)
   end
 
+  def include_thumbnails?
+    ThemeModifierHelper.new(request: scope.request).topic_thumbnail_sizes.present? || DiscoursePluginRegistry.topic_thumbnail_sizes.present?
```

Comment:
In `include_excerpt?` we're already creating an instance of `ThemeModifierHelper`, perhaps we can memoize the instance of `ThemeModifierHelper` and reuse that here?

Reply:
We could, but `ThemeModifierHelper` is [extremely lightweight](https://github.com/discourse/discourse/blob/master/lib/theme_modifier_helper.rb#L3-L5), so I don't think it should make much difference?

Reply:
Ooops sorry just saw that `include_excerpt?` doesn't always create the instance of `ThemeModifierHelper`.